src/python/pants/backend/experimental/python/lockfile_metadata.py

```diff
@@ -32,34 +36,47 @@ def from_lockfile(cls, lockfile: bytes) -> LockfileMetadata:
             elif in_metadata_block:
                 metadata_lines.append(line[2:])
 
+        if not metadata_lines:
+            # TODO(#12314): Add a good error.
+            raise InvalidLockfileError("")
+
         try:
             metadata = json.loads(b"\n".join(metadata_lines))
         except json.decoder.JSONDecodeError:
             # TODO(#12314): Add a good error.
-            raise
-
-        T = TypeVar("T")
+            raise InvalidLockfileError("")
 
-        def coerce(t: Callable[[Any], T], key: str) -> T | None:
-            """Gets a value from `metadata`, coercing it to type `t` if not `None`."""
-            v = metadata.get(key, None)
+        def get(key: str) -> Any:
             try:
-                return t(v) if v is not None else None
-            except Exception:
-                # TODO: this should trigger error/warning behavior
-                return None
-
-        return LockfileMetadata(
-            requirements_invalidation_digest=coerce(str, "requirements_invalidation_digest"),
-            valid_for_interpreter_constraints=coerce(
-                InterpreterConstraints, "valid_for_interpreter_constraints"
-            ),
-        )
+                return metadata[key]
+            except KeyError:
+                # TODO(#12314): Add a good error.
+                raise InvalidLockfileError("")
+
+        digest = get("requirements_invalidation_digest")
```

Comment:
Given the rest of the discussions we've had about confusion around this name, it's probably best that this is called `hex_digest`